src/modules/flight_mode_manager/tasks/AutoFollowTarget/FlightTaskAutoFollowTarget.hpp

```diff
@@ -125,91 +163,190 @@ class FlightTaskAutoFollowTarget : public FlightTask
+	 */
+	void update_target_orientation(float &current_target_orientation, const Vector2f &target_velocity, const Vector2f &target_velocity_unfiltered);
+
+	/**
+	 * Updates the orbit angle setpoint, taking into account the maximal orbit tangential speed
+	 *
+	 * Returns the orbit angle setpoint, taking into account the maximal orbit tangential speed.
+	 * While setting the orbital velocity setpoint vector for the according position setpoint
+	 *
+	 * @param target_orientation Tracked target orientation
+	 * @param previous_orbit_angle_setpoint Previous orbit angle setpoint
+	 *
+	 * @return Angle [rad] Next feasible orbit angle setpoint
 	 */
-	matrix::Vector3f predict_future_pos_ned_est(float deltatime, const matrix::Vector3f &pos_ned_est,
-			const matrix::Vector3f &vel_ned_est,
-			const matrix::Vector3f &acc_ned_est) const;
+	float update_orbit_angle_trajectory(const float target_orientation, const float previous_orbit_angle_setpoint);
 
+	/**
+	 * Returns the orbit angle setpoint, taking into account the maximal orbit tangential speed.
+	 *
+	 * @param orbit_angle_setpoint Orbit angle setpoint
+	 *
+	 * @return 2D Velocity Vector [m/s, m/s] of current orbit position setpoint
+	*/
+	Vector2f get_orbit_tangential_velocity(const float orbit_angle_setpoint) const;
+
+	/**
+	 * Calculates desired drone position taking into account orbit angle and the follow target altitude mode
+	 *
+	 * @param target_position Tracked target position Vector3f reference
+	 * @param orbit_angle_setpoint Current orbit angle setpoint around the target
+	 *
+	 * @return Position [m,m,m] Final position setpoint for the drone
+	 */
+	Vector3f calculate_desired_drone_position(const Vector3f &target_position, const float orbit_angle_setpoint) const;
+
+	/**
+	 * Calculate the gimbal height offset to the target to calculate the pitch angle command
+	 *
+	 * @param altitude_mode Current Follow Target Altitude mode
+	 * @param target_pos_z Target's local position z value to use for 3D tracking
+	 *
+	 * @return Height [m] Difference between the target and the drone
+	 */
+	float calculate_gimbal_height(const FollowAltitudeMode altitude_mode, const float target_pos_z) const;
+
+	/**
+	 * Publishes gimbal control command to track the target, given xy distance and z (height) difference
+	 *
+	 * @param xy_distance Horizontal distance to target
+	 * @param z_distance Vertical distance to target
+	 *
+	 * @return Angle [rad] Gimbal pitch setpoint, for logging in follow_target_status uORB message
+	 */
+	float point_gimbal_at(const float xy_distance, const float z_distance);
+
+	/**
+	 * Get the current follow-me perspective angle setting from PX4 parameters
+	 *
+	 * @param follow_perspective value of the parameter FLW_TGT_FS
+	 * @return Angle [deg] from which the drone should view the target while following it, with zero degrees indicating the target's 12 o'clock
+	 */
+	float get_follow_angle_setting_deg(const FollowPerspective follow_perspective) const;
```

Comment:
see previous comment.. i think this method can be removed

Reply:
Yup!

Reply:
Ooh but I think that if a user sets an invalid parameter, it could access the wrong index of the array of angles. Therefore having a safety net function to catch cases like this would be better in my opinion.

Reply:
what about something like this?
```
float convertFollowPerspectiveToRadians(const FollowPerspective follow_perspective) const
{
	const float follow_angle_rad = (follow_perspective < kFollowPerspectiveInvalid) ? _follow_perspective_angles[follow_perspective] : kFollowPerspectiveBehind;
	return math::radians(follow_angle_rad);
}
```
where, in class declaration:
```
static constexpr float kFollowPerspectiveAnglesDeg[xx] = {0.0f, 45.0f, ....};
```

Reply:
Yes I thought the function name is a bit too vague, I like this better!